code/modules/shuttle/spaceship_navigation_beacon.dm

```diff
@@ -0,0 +1,78 @@
+/obj/item/circuitboard/machine/spaceship_navigation_beacon
+	name = "Bluespace Navigation Gigabeacon (Machine Board)"
+	build_path = /obj/machinery/spaceship_navigation_beacon
+	req_components = list()
+
+
+/obj/machinery/spaceship_navigation_beacon
+
+	name = "Bluespace Navigation Gigabeacon"
+	desc = "A device that creates a bluespace anchor that allow ships jump near to it."
+
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "floor_beaconf"
+
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 0
+
+	var/id = ""
+	var/access_code = ""
+
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/spaceship_navigation_beacon
+
+
+/obj/machinery/spaceship_navigation_beacon/Initialize()
+	. = ..()
+	SSshuttle.beacons |= src
+
+/obj/machinery/spaceship_navigation_beacon/random/Initialize()
+	.=..()
+	randomise_beacon()
+
+/obj/machinery/spaceship_navigation_beacon/proc/randomise_beacon()
+	id = "[num2hex(rand(1,65535), -1)]" //gives us a random four-digit hex number as id.
```

Comment:
You have the possibility of overlapping ids even if it is tiny, do this instead
```
var/static/newid = rand(1, 65535)
newid = WRAP(newid + 1, 1, 65535)
id = num2hex(newid, -1)
```

Reply:
or, use the GUID proc that already exists